examples/social/src/contract.rs

```diff
@@ -80,9 +92,21 @@ impl Contract for SocialContract {
 }
 
 impl SocialContract {
-    async fn execute_post_operation(&mut self, text: String) -> (Destination, Message) {
+    async fn execute_post_operation(
+        &mut self,
+        text: String,
+        image_url: Option<String>,
+    ) -> (Destination, Message) {
         let timestamp = self.runtime.system_time();
-        self.state.own_posts.push(OwnPost { timestamp, text });
+        let image = match image_url {
+            Some(image) => Some(image),
+            None => None,
+        };
```

Comment:
Our lint checks are unhappy about this `match` because it's a no-op :)

Reply:
the post expects an Option<String> type. but this match statement has no use because i am assigning variable image as an Option. if user does not provide a image_url then it should set it to `None`. 
but yeah having an option type will automatically set it to None if its not received. :)